helpers/compile/plugins/replaceWithPlugin.ts

```diff
@@ -1,27 +1,39 @@
 import type * as esbuild from 'esbuild'
 import fs from 'fs'
 
-function applyReplacements(contents: string, replacements: [RegExp, string][]) {
+type Replacement = [RegExp, string | ((regex: RegExp, contents: string) => string | Promise<string>)]
+
+async function applyReplacements(contents: string, replacements: Replacement[]) {
   for (const [regex, replacement] of replacements) {
-    contents = contents.replace(regex, replacement)
+    if (typeof replacement === 'string') {
+      contents = contents.replace(regex, replacement)
+    } else {
+      contents = await replacement(regex, contents)
+    }
   }
 
   return contents
 }
 
+// native nodejs imports so that we can filter out
+const nativeDependencies = new Set(Object.keys((process as any).binding('natives')))
```

Comment:
How does this work?

Reply:
I can't remember where I found this, but it is an internal API that is used all over the Node.js codebase for importing native libraries from C++ if I remember well. The param is a selector to get a lib (eg. `binding('fs')`) or filter all of them with `natives` (for Node's libraries). You'd get something like:
```
  '_http_agent',
  '_http_client',
  '_http_common',
  '_http_incoming',
  '_http_outgoing',
  '_http_server',
  '_stream_duplex',
  '_stream_passthrough',
  '_stream_readable',
  '_stream_transform',
  '_stream_wrap',
  '_stream_writable',
  '_tls_common',
  '_tls_wrap',
  'assert',
  'async_hooks',
  'buffer',
  'child_process',
  'cluster',
  'console',
  'constants',
  'crypto',
  'dgram',
  'dns',
  'domain',
  'events',
  'fs',
  'fs/promises',
  'http',
  'http2',
  'https',
```

Reply:
Maybe worth looking at: https://github.com/nodejs/node/issues/3307

Reply:
Thanks a lot, I'll give a try to https://nodejs.org/api/module.html#modulebuiltinmodules. Looks more robust.